src/components/schedule-page/table/schedule/sections/base-section/base-section.component.tsx

```diff
@@ -65,29 +68,80 @@ function BaseSectionComponentF({
       setPointerPosition(newPosition);
     }
   }
-  function resetPointer(): void {
+
+  function resetSelection(): void {
     setPointerPosition({ row: -1, cell: -1 });
+    setSelectionMatrix(getCopy(selectionMatrix));
   }
+
+  const [selectionMatrix, setSelectionMatrix] = useState(
+    data.map((c) => c.rowData().map((_) => false))
+  );
+
+  function onDrag(pivot: PivotCell, rowInd: number, cellInd: number): void {
+    setSelectionMatrix((prev) =>
+      setSelectionSquare(prev, pivot.cellIndex, pivot.rowIndex, cellInd, rowInd)
+    );
+  }
+
+  function getCopy(source: SelectionMatrix): SelectionMatrix {
+    return [...Array(source.length)].map((_) => Array(source[0].length).fill(false));
+  }
+
+  const setSelectionSquare = useCallback(
+    (source: SelectionMatrix, x1: number, y1: number, x2: number, y2: number): SelectionMatrix => {
+      const selection = getCopy(source);
+      const [startX, endX] = x2 < x1 ? [x2, x1] : [x1, x2];
+      const [startY, endY] = y2 < y1 ? [y2, y1] : [y1, y2];
+      if (startX < 0 || startY < 0) {
+        return selection;
+      }
+      for (let y = startY; y <= endY; ++y) {
+        for (let x = startX; x <= endX; ++x) {
+          selection[y][x] = true;
+        }
+      }
+      return selection;
+    },
+    []
```

Comment:
Logika która odpowiada za to żeby zaznaczyć kwadrat z wierzchołkami w (x1, y1), (x2,y2).
Zastanawiałem się nad czy nie wynieść tego do helperu, ale postanowiłem, że lepiej nie, bo nie będzie to używane.

Ale już mam świetny pomysł na refaktoring:
to można napisać za pomocą `hook'a`.
I tak będzie o wiele lepiej